### PR TITLE
feat: added emulation of virtual machine movements

### PR DESCRIPTION
**Pull request by Isteb4k**
## Description

Shatal - the tool for virtual machine wobbling.

Available operations:
- Node draining with migration of virtual machines
- Update core fraction from 10% to 25% and vice versa
- Creation of virtual machines
- Deletion of virtual machines

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

**Review by yaroslavborbat — APPROVED** [review]

File: tests/performance/shatal/internal/shatal/drainer.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				if sortedNodes[i] == s.lastDrainedNode {
+					node = sortedNodes[i+1]
+					break
+				}
+			}
+
+			s.nodes[node].Lock()
+			s.logger.Info(fmt.Sprintf("Drain: %s", node))
+
+			cmd := exec.CommandContext(context.Background(), "kubectl", "drain", node, "--ignore-daemonsets", "--delete-emptydir-data")
```
Comment (yaroslavborbat) [inline tests/performance/shatal/internal/shatal/drainer.go:70]: conflicts with kubeconfigBase64 in config

File: tests/performance/shatal/internal/shatal/drainer.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+				cmd.Stdout = os.Stdout
+			}
+
+			err := cmd.Run()
+			if err != nil {
+				s.logger.Error(err.Error())
+				return
+			}
+
+			cmd = exec.CommandContext(context.Background(), "kubectl", "uncordon", node)
```
Comment (yaroslavborbat) [inline tests/performance/shatal/internal/shatal/drainer.go:82]: conflicts with kubeconfigBase64 in config

**Review by yaroslavborbat — CHANGES_REQUESTED** [review]

File: tests/performance/shatal/Taskfile.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# https://taskfile.dev
+
+version: "3"
+
+silent: true
+
+tasks:
+  run:
+    desc: "Run emulation of virtual machine movements"
```
Comment (universal-itengineer) [inline tests/performance/shatal/Taskfile.yaml:9]:
```suggestion
    desc: "Run of virtual machine migrations after node drain"
```

File: tests/performance/templates/vms.yaml [hunk 4]
```diff
@@ -12,6 +12,12 @@ metadata:
 spec:
   runPolicy: AlwaysOn
   enableParavirtualization: true
+  disruptions:
+    {{- if eq (mod . 2) 0 }}
+    restartApprovalMode: Automatic
+    {{- else }}
+    restartApprovalMode: Manual
+    {{- end }}
```
Comment (diafour) [inline tests/performance/templates/vms.yaml:20]: Comment this block please. It is not clear why Automatic/Manual should depend on VM number.

File: tests/performance/shatal/internal/shatal/modifier.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		namespace: namespace,
+		logger:    log.With("type", "modifier"),
+	}
+}
+
+func (s *Modifier) Do(ctx context.Context, vm v1alpha2.VirtualMachine) {
+	if vm.Spec.CPU.CoreFraction == "10%" {
+		vm.Spec.CPU.CoreFraction = "25%"
+	} else {
+		vm.Spec.CPU.CoreFraction = "10%"
```
Comment (diafour) [inline tests/performance/shatal/internal/shatal/modifier.go:36]: Move them to constants

File: tests/performance/shatal/internal/shatal/modifier.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	s.logger.
+		With("node", vm.Status.NodeName).
+		With("core-fraction", vm.Spec.CPU.CoreFraction).
+		Info(fmt.Sprintf("Modify: %s", vm.Name))
+
+	err := s.api.UpdateVM(ctx, vm)
```
Comment (diafour) [inline tests/performance/shatal/internal/shatal/modifier.go:44]: I have a feeling that Patch is better, as Update may lead to error "unable to update ... the object has been modified; please apply your changes to the latest version and try again".

File: tests/performance/shatal/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+## Available operations
+
+- Node draining with migration of virtual machines
+- Update core fraction from 10% to 25% and vice versa
+- Creation of virtual machines
+- Deletion of virtual machines
+## How to run
+
+```shell
+task run
```
Comment (universal-itengineer) [inline tests/performance/shatal/README.md:12]:
```suggestion
KUBECONFIG=$(cat ~/.kube/config | base64 -w 0)
KUBECONFIG_BASE64=$KUBECONFIG task shatal:run
```

**Review by universal-itengineer — APPROVED** [review]

**Review by yaroslavborbat — APPROVED** [review]

**Pull request merged**